pkgs/by-name/sw/switch-to-configuration-ng/src/src/main.rs

```diff
@@ -1714,15 +1688,12 @@ won't take effect until you reboot the system.
 
     // Reload units that need it. This includes remounting changed mount units.
     if !units_to_reload.is_empty() {
-        let mut units = units_to_reload
-            .keys()
-            .into_iter()
-            .map(String::as_str)
-            .collect::<Vec<&str>>();
-        units.sort_by_key(|name| name.to_lowercase());
-        eprintln!("reloading the following units: {}", units.join(", "));
-
-        for unit in units {
+        eprintln!(
+            "reloading the following units: {}",
+            format_unit_set(&units_to_reload)
+        );
+
+        for unit in &units_to_reload {
```

Comment:
FWIW, this is a *slight* change in behavior, as this `for` loop is no longer iterating in a sorted order over `to_lowercase`'d unit names. I don't *think* this matters? Might want to give this a second thought though.

Reply:
Yes, that's true. I was considering changing the set to be a BTreeSet, which would make all behavior deterministic. What do you think?

Reply:
Can you explain how that restores the previous behavior?

To be clear, I'm not married to the previous behavior. I just wanted to note that this is different.

Reply:
Yes, I should have pointed this change in semantics out more clearly.

As opposed to HashSets, [BTreeSets](https://doc.rust-lang.org/std/collections/struct.BTreeSet.html) are deterministic for the purposes of iterating over them, because they produce elements in order:

> Iterators returned by BTreeSet::iter and BTreeSet::into_iter produce their items in order, and take worst-case logarithmic and amortized constant time per item returned.

This does not restore the previous behavior, but it will result in deterministic behavior (i.e. the units will be reloaded in the same order every time).

If we are not married to the lowercase ordering when printing units that need to be restarted/etc, we can also remove the manual sorting step and further simplify the code. But maybe this is for another day. I don't want to shave the yak here. :smile: